RMC Welder values and satchels.

Thereisabear opened the pull request:
## About the PR
Adjusts the fuel quantity in welders and welding bags/satchels
Adjusts the passive drain when blowtorch is lit. 
Sets fuel cost on the construction graphs for most things that require welding.
All values taken from CM.

Small RMCWeldingSystem to intercept upstream costs on welding shut/open things.

## Why / Balance
Parity

## Media
https://github.com/user-attachments/assets/13f7e288-e943-43dd-b346-999fc91ac828

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->
- [X] By submitting this code and/or assets, I confirm that I either own them or have provided the correct necessary licenses to use and distribute them. I agree to be fully responsible for any legal claims or issues arising from the use of these materials.

**Changelog**

:cl:

- Fix: Fixed blowtorches using too much fuel when deconstructing walls or reinforced tables.
- Tweak: Welding satchels & bags now hold their intended capacities. 
- Tweak: Welders now have their intended fuel usage, storage and passive lit cost.

Thereisabear committed:
Welder Parity

Welder Parity

Thereisabear committed:
Teleporting tools fix

Thereisabear committed:
changed wrong tank capacity

DrSmugleaf commented on Content.Shared/Tools/Systems/SharedToolSystem.Welder.cs:
```diff
@@ -191,9 +191,38 @@ private void OnActivateAttempt(Entity<WelderComponent> entity, ref ItemToggleAct
     {
-        if (args.User != null && !_actionBlocker.CanComplexInteract(args.User.Value)) {
+        if (args.User != null && !_actionBlocker.CanComplexInteract(args.User.Value))
+        {
```
The styling is incorrect but it's better not to fix styling for upstream files, for conflicts

```suggestion
        if (args.User != null && !_actionBlocker.CanComplexInteract(args.User.Value)) {
```

DrSmugleaf commented on Content.Shared/Tools/Systems/SharedToolSystem.Welder.cs:
```diff
@@ -191,9 +191,38 @@ private void OnActivateAttempt(Entity<WelderComponent> entity, ref ItemToggleAct
         }
     }
+
+    /// Amount of 0 checks if fuel exists without consuming.
```
```suggestion
    // RMC14
    /// Amount of 0 checks if fuel exists without consuming.
```

Thereisabear committed:
Update Content.Shared/Construction/Steps/ToolConstructionGraphStep.cs

Co-authored-by: DrSmugleaf <10968691+DrSmugleaf@users.noreply.github.com>

Thereisabear committed:
Update Content.Shared/Placeable/PlaceableSurfaceSystem.cs

Co-authored-by: DrSmugleaf <10968691+DrSmugleaf@users.noreply.github.com>

Thereisabear committed:
Update Content.Shared/Tools/Components/WeldableComponent.cs

Co-authored-by: DrSmugleaf <10968691+DrSmugleaf@users.noreply.github.com>

Thereisabear committed:
Update Content.Shared/Tools/Components/WelderComponent.cs

Co-authored-by: DrSmugleaf <10968691+DrSmugleaf@users.noreply.github.com>

Thereisabear committed:
Update Content.Shared/Tools/Components/WelderComponent.cs

Co-authored-by: DrSmugleaf <10968691+DrSmugleaf@users.noreply.github.com>

Thereisabear committed:
Update Content.Shared/Tools/Systems/SharedToolSystem.Welder.cs

Co-authored-by: DrSmugleaf <10968691+DrSmugleaf@users.noreply.github.com>

Thereisabear committed:
Update Content.Shared/Tools/Systems/SharedToolSystem.Welder.cs

Co-authored-by: DrSmugleaf <10968691+DrSmugleaf@users.noreply.github.com>

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
Update welders.yml

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
Isolated intentions to RMC specific components.

Because upstream merge conflicts are headaches.

Thereisabear commented:
Code has been altered so it now does not edit any upstream files, as I realised fuelcost can just be set by RMCRepairable fuelUsed:  and RMCConstruction thingy

Thereisabear committed:
Update RMCInputSystem.cs

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
Merge branch 'master' into welderfuelusage

DrSmugleaf commented on Content.Shared/_RMC14/Input/RMCInputSystem.cs:
```diff
@@ -14,18 +14,34 @@ public sealed class RMCInputSystem : EntitySystem
+            if (mover.RelativeEntity == ev.Entity)
+                mover.RelativeEntity = null;
+        }
+    }
```
This should be directed at a component, otherwise this will run any time any entity in the game deletes, including bullets
If there's no existing component for this you would need to add a new one

Thereisabear replied:
Should be resolved, I hooked it onto map and mapgrid comps as from what i can tell? 
They the only one that use the relativeentity.

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
comp

Thereisabear committed:
subscribed to map/mapgridcomp

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
terminating and dictionary fix

Thereisabear committed:
clean up

Thereisabear committed:
more cleanup

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
welderbag/packs/satchels fixes

DrSmugleaf commented on Content.Shared/_RMC14/Input/RMCInputSystem.cs:
```diff
@@ -23,9 +24,32 @@ public override void Initialize()
+        while (query.MoveNext(out _, out var mover))
+        {
+            if (mover.RelativeEntity == uid)
+                mover.RelativeEntity = null;
```
This needs to be Dirty'd as well so it syncs to the client
You would need to use the entity id from the query itself

Thereisabear replied:
Done

Thereisabear committed:
dirtied

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
Update welders.yml

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
Update welders.yml

Thereisabear committed:
Update welders.yml

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
Merge branch 'master' into welderfuelusage

Thereisabear committed:
Split

Thereisabear committed:
Split 2.

I actually forgot what this fix was for.

Thereisabear committed:
Name Shortened.

Thereisabear committed:
Delete RMCBlowtorchWeldFuelComponent.cs

PursuitInAshes reviewed (dismissed):
LGTM
Thanks for the PR

Thereisabear committed:
fix to fix a strange compound

though realistically wouldnt happen in live but better safe than sorry i guess.

Thereisabear commented:
Final and ready

kleinerstation13 reviewed (approved):
I can't comment on the C# but yml is fine, parity checked the values and it looks correct
